dev_tools/nyaml2nxdl/nyaml2nxdl_backward_tools.py

```diff
@@ -0,0 +1,1046 @@
+                    f"The allowed attr is {allowed_attr}."
+                )
+
+        has_min_max = False
+        has_opt_reco_requ = False
+        if exists_dict:
+            for key, val in exists_dict.items():
+                if key in ["minOccurs", "maxOccurs"]:
+                    tmp_dict["exists"] = tmp_dict["exists"] + val
+                    has_min_max = True
+                elif key in ["optional", "recommended", "required"]:
+                    tmp_dict["exists"] = key
+                    has_opt_reco_requ = True
+        if has_min_max and has_opt_reco_requ:
+            raise ValueError(
+                "Optionality 'exists' can take only either from ['minOccurs',"
+                " 'maxOccurs'] or from ['optional', 'recommended', 'required']"
+                ". But not from both of the groups together. Please check in"
+                " attributes"
+            )
+
+        depth_ = depth + 1
+        for key, val in tmp_dict.items():
+            # Increase depth size inside handle_map...() for writting text with one
+            # more indentation.
+            file_out.write(
+                f"{depth_ * DEPTH_SIZE}{key}: "
+                f"{handle_mapping_char(val, depth_ + 1, False)}\n"
+            )
+
+    def handel_link(self, depth, node, file_out):
+        """
+        Handle link elements of nxdl
+        """
+
+        possible_link_attrs = ["name", "target", "napimount"]
+        node_attr = node.attrib
+        # Handle special cases
+        if "name" in node_attr:
+            file_out.write(
+                "{indent}{name}(link):\n".format(
+                    indent=depth * DEPTH_SIZE, name=node_attr["name"] or ""
+                )
+            )
+            del node_attr["name"]
+
+        depth_ = depth + 1
+        # Handle general cases
+        for attr_key, val in node_attr.items():
+            if attr_key in possible_link_attrs:
+                file_out.write(
+                    "{indent}{attr}: {value}\n".format(
+                        indent=depth_ * DEPTH_SIZE, attr=attr_key, value=val
+                    )
+                )
+            else:
+                raise ValueError(
+                    f"An anexpected attribute '{attr_key}' of link has found."
+                    f"At this moment the alloed keys are {possible_link_attrs}"
+                )
+
+    def handel_choice(self, depth, node, file_out):
+        """
+        Handle choice element which is a parent node of group.
+        """
+
+        possible_attr = []
+
+        node_attr = node.attrib
+        # Handle special casees
+        if "name" in node_attr:
+            file_out.write(
+                "{indent}{attr}(choice): \n".format(
+                    indent=depth * DEPTH_SIZE, attr=node_attr["name"]
+                )
+            )
+            del node_attr["name"]
+
+        depth_ = depth + 1
+        # Taking care of general attrinutes. Though, still no attrinutes have found,
```

Comment:
"Take care of general attributes. Note other choices may be allowed in the future"